components/dashboard/src/projects/ProjectVariables.tsx

```diff
@@ -0,0 +1,126 @@
+                <button onClick={() => setShowAddVariableModal(true)}>New Variable</button>
+            </div>
+            : <>
+                <ItemsList>
+                    <Item header={true} className="grid grid-cols-5 items-center">
+                        <ItemField>Name</ItemField>
+                        <ItemField>Value</ItemField>
+                        <ItemField>In Prebuilds?</ItemField>
+                        <ItemField>In Workspaces?</ItemField>
+                        <ItemField></ItemField>
+                    </Item>
+                    {envVars.map(variable => {
+                        return <Item className="grid grid-cols-5 items-center">
+                            <ItemField>{variable.name}</ItemField>
+                            <ItemField>****</ItemField>
+                            <ItemField>Visible</ItemField>
+                            <ItemField>{variable.censored ? 'Censored' : 'Visible'}</ItemField>
+                            <ItemField className="text-right"><button onClick={() => deleteEnvVar(variable.id)}>x</button></ItemField>
+                        </Item>
+                    })}
+                </ItemsList>
+            </>
+        }
+    </ProjectSettingsPage>;
+}
+
+function AddVariableModal(props: { project?: Project, onClose: () => void }) {
+    const [ name, setName ] = useState<string>("");
+    const [ value, setValue ] = useState<string>("");
+    const [ censored, setCensored ] = useState<boolean>(false);
+    const [ error, setError ] = useState<Error | undefined>();
+
+    const addVariable = async () => {
+        if (!props.project) {
+            return;
+        }
+        try {
+            await getGitpodService().server.setProjectEnvironmentVariable(props.project.id, name, value, censored);
+            props.onClose();
+        } catch (err) {
+            setError(err);
+        }
+    }
+
+    return <Modal visible={true} onClose={props.onClose} onEnter={() => { addVariable(); return false; }}>
+        <h3 className="mb-4">Add Variable</h3>
+        <div className="border-t border-b border-gray-200 dark:border-gray-800 -mx-6 px-6 py-4 flex flex-col">
+            {error && <div className="bg-gitpod-kumquat-light rounded-md p-3 text-gitpod-red text-sm mb-2">
+                {error}
+            </div>}
+            <div>
+                <h4>Name</h4>
+                <input autoFocus className="w-full" type="text" name="name" value={name} onChange={e => setName(e.target.value)} />
+            </div>
+            <div className="mt-4">
+                <h4>Value</h4>
+                <input className="w-full" type="text" name="value" value={value} onChange={e => setValue(e.target.value)} />
+            </div>
+            <div className="mt-4">
+                <CheckBox title="Visible in Prebuilds?" desc="All Project Variables are visible in Prebuilds" checked={true} disabled={true} />
```

Comment:
**issue**: Including always disabled checkboxes does not help much, right? I'd suggest to drop this option and replace it with a warning alert. What do you think?

Reply:
I agree that it's a "placebo" checkbox, but in this case I felt like it really helps clarify where variables are exposed and what you can change about it.

Alternatives like having a single checkbox, and a long complicated text to explain the full situation, seem less intuitive, i.e. it requires more upfront study work & cognitive overheard for users before they can fully grasp this. A "fake" checkbox that you immediately understand seemed like the lesser evil, but I'd be curious if you have ideas to make users immediately understand the situation without needing a "fake" checkbox?

Reply:
Thanks, Jan! Let's do this.

![60taoh](https://user-images.githubusercontent.com/120486/148917968-6dc32855-bb16-45b0-a56e-29aadba77f20.jpg)